java/arcs/core/entity/Handle.kt

```diff
@@ -53,13 +59,78 @@ suspend fun <T : Handle> T.awaitReady(): T = suspendCancellableCoroutine<T> { co
 interface Storable
 
 /** Configuration for a [Handle]. */
-data class HandleSpec<T : Entity>(
+data class HandleSpec(
     val baseName: String,
     val mode: HandleMode,
-    val containerType: HandleContainerType,
-    val entitySpec: EntitySpec<T>,
-    val dataType: HandleDataType = HandleDataType.Entity
-)
+    val type: Type,
+    val entitySpecs: Set<EntitySpec<*>>
+) {
+
+    // This Data Class is in the midst of refactoring from only supporting handles storing
+    // a single top-level entity, e.g. handles of types: Product, &Product, [Product], [&Product]
+    // To allowing handles that stores objects having multiple top level entities,
+    // e.g. a handle of type: [(Product, Review, Manufacturer)]
+    //
+    // Everything below this comment is a compatibility layer for code that expects the former,
+    // and it should gradually be disappearing as we clean things up.
```

Comment:
These kinds of comments have a habit of sticking around. It might be better to add `@deprecated` annotations to everything

Reply:
Deprecated the constructor and the entitySpec getter, but left the containerType and innerType for now - unclear whether we will be removing it for now.